### PR TITLE
update compatibility tests with new release branches

### PR DESCRIPTION
**Pull request by crodriguezvega**
## Description

There's new release branches v4.3.x, v5.2.x and v6.1.x. v5.0.x, v5.1.x and v6.0.x will become unmaintained, so I am removing them from the compatibility tests.

closes: #XXXX

### Commit Message / Changelog Entry

```bash
type: commit message
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/CONTRIBUTING.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/10-structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/CONTRIBUTING.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [x] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

**Comment by crodriguezvega** [comment]
- [Compatibility test run](https://github.com/cosmos/ibc-go/actions/runs/3711515990) for v6.1.x
- [Compatibility test run](https://github.com/cosmos/ibc-go/actions/runs/3711518350) for v5.2.x

File: .github/compatibility-test-matrices/release-v4.3.x/incentivized-transfer-chain-a.json
Comment (damiannolan) [inline .github/compatibility-test-matrices/release-v4.3.x/incentivized-transfer-chain-a.json:3]: Do we still need to include retracted release tags?

**Review by damiannolan — APPROVED** [review]
🙏

File: .github/compatibility-test-matrices/release-v4.3.x/incentivized-transfer-chain-a.json
Comment (crodriguezvega) [reply .github/compatibility-test-matrices/release-v4.3.x/incentivized-transfer-chain-a.json:3]: I wanted to leave them here to run the compatibility tests and then remove them once we tag the new releases.

**Review by chatton — APPROVED** [review]
LGTM! Happy to merge once we merge #2949 into this!

**Pull request merged**
